backend/FwLite/MiniLcm/SyncHelpers/EntrySync.cs

```diff
@@ -0,0 +1,124 @@
+            },
+            //do nothing, complex form types are not editable, ignore any changes to them here
+            static (api, beforeComplexFormType, afterComplexFormType) => Task.FromResult(0));
+    }
+
+    private static async Task<int> Sync(IList<ComplexFormComponent> afterComponents, IList<ComplexFormComponent> beforeComponents, IMiniLcmApi api)
+    {
+        return await DiffCollection.Diff(api,
+            beforeComponents,
+            afterComponents,
+            //we can't use the ID as there's none defined by Fw so it won't work as a sync key
+            component => (component.ComplexFormEntryId, component.ComponentEntryId, component.ComponentSenseId),
+            static async (api, afterComponent) =>
+            {
+                //change id, since we're not using the id as the key for this collection
+                //the id may be the same, which is not what we want here
+                afterComponent.Id = Guid.NewGuid();
+                await api.CreateComplexFormComponent(afterComponent);
+                return 1;
+            },
+            static async (api, beforeComponent) =>
+            {
+                await api.DeleteComplexFormComponent(beforeComponent);
+                return 1;
+            },
+            static async (api, beforeComponent, afterComponent) =>
+            {
+                if (beforeComponent.ComplexFormEntryId == afterComponent.ComplexFormEntryId &&
+                    beforeComponent.ComponentEntryId == afterComponent.ComponentEntryId &&
+                    beforeComponent.ComponentSenseId == afterComponent.ComponentSenseId)
+                {
+                    return 0;
+                }
+                await api.ReplaceComplexFormComponent(beforeComponent, afterComponent);
+                return 1;
+            }
+        );
+    }
+
+    private static async Task<int> SensesSync(Guid entryId,
+        IList<Sense> afterSenses,
+        IList<Sense> beforeSenses,
+        IMiniLcmApi api)
+    {
+        Func<IMiniLcmApi, Sense, Task<int>> add = async (api, afterSense) =>
+        {
+            await api.CreateSense(entryId, afterSense);
+            return 1;
+        };
+        Func<IMiniLcmApi, Sense, Task<int>> remove = async (api, beforeSense) =>
+        {
+            await api.DeleteSense(entryId, beforeSense.Id);
+            return 1;
+        };
+        Func<IMiniLcmApi, Sense, Sense, Task<int>> replace = async (api, beforeSense, afterSense) => await SenseSync.Sync(entryId, afterSense, beforeSense, api);
+        return await DiffCollection.Diff(api, beforeSenses, afterSenses, add, remove, replace);
+    }
+
+    public static UpdateObjectInput<Entry>? EntryDiffToUpdate(Entry beforeEntry, Entry afterEntry)
+    {
+        JsonPatchDocument<Entry> patchDocument = new();
+        patchDocument.Operations.AddRange(MultiStringDiff.GetMultiStringDiff<Entry>(nameof(Entry.LexemeForm), beforeEntry.LexemeForm, afterEntry.LexemeForm));
```

Comment:
What are your thoughts on using reflection for this?

Reply:
it's certainly possible, probably how I would have done it in the past, but I'd rather just do this and have tests catch when we miss something, maybe once we have more objects it'll be a bigger burden, but it makes it much easier to read